### DISCUSSION TITLE
Airflow temperature sensor

### DISCUSSION THREAD

**Discussion by FluffyPotatoz** [DIY-Flow-Bench/DIY-Flow-Bench#160] [General]
Hi, hopefully you don't mind me asking questions as i couldn't find anywhere else to discuss the build of this project.

I am in the process of building a flow bench, and decided to take the MAF route instead of the differential orifice route for my first bench build. One of the things i have a question is regarding additional air temperature sensors in the plenum after the "head" to be measured. I realize that we are using a BME280 which is reading ambient air values, and that works fine for vacuum measuring intake valve flow. However, don't we want a air temp sensor in the air plenum when measuring exhaust valve blow thru temperature (after its heated up by reversing the flow of the vacuum generator source)?. Wouldn't this cause a change in air density values that the MAF cant compensate for? My research on MAF's are not up to date lol. I have used PT1000/MAX31965 breakouts for other projects and pondering if this is a option to use this breakout on this setup. Are there provisions on the SPI bus to allow this?

Also, i am looking at the suggested 0280218067 Bosch MAP pin out, and it looks like it has an integrated temp sensor. Are there previsions in the code to hook up and enable reading from this sensor. Obviously i would assume PT1000's are more accurate then the built in sensor, and would be preferred.

Or, does the MAF compensate for air temp internally, and its output value is already compensated?. If that's the case, my question is answered.

Thanks

**Comment by DeeEmm** [comment]
Always happy to answer questions if I can.

You raise an interesting point. The MAF itself, is temperature compensated, so the value measured will always be accurate, but to convert from MASS to volumetric requires the additional environmental data from the BME to determine the air density

Your observation that reverse air flow may be heated by the motor is valid, and a factor I recall reading about when using traditional benches. 

Solutions include 
- Mounting the BME in the plenum (if the bench design allows it). This way the MAF and the BME will see the same air. 
- Providing a fixed compensation offset - I believe that this is how the problem was dealt with on traditional benches. 
- Using an additional temperature sensor as suggested. 
- Reversing the flow through the head so that air is drawn from the exhaust outlet instead of blown from the valve side

There is currently no provision to use the Temp sensor from the MAF as not all MAF sensors provide this. Also as far as I can recall there is no temperature compensation other than that used from the BME.

The current design does not provide breakout pins for the IC2 bus, but these would be easy to provision. There was at one point provision for PT1000's in the code but I would need to check if this is still present.

I would propose that the best solution is to add temperature compensation to the code which supports various input sources (BME / MAF / PT1000 / Analog / I2C / etc) as well as a fixed offset. This way the end user can decide what is appropriate

I will add this to the V3 roadmap.

**Reply by FluffyPotatoz** [reply]
Fantastic Thanks for the reply!

I think what i might do in the mean time, is order 2 of the BME modules and install one inside the plenum for temp compensation when doing blowthru, and just do a simple plug and play cable switch to select the appropriate BME module depending on the use case. obviously there would be some "changeover" time required when swapping from suckthru to blowthru, ie swapping the direction of the MAF sensor. Another alternative could be placing the MAF on the suction side of the "vacuum generator" in blowthru mode if the system is guaranteed to never have any leaks, i guess in theory the "mass" air flow into the vacuums should be the same as the output thru into the head. Im curious to test and see what orientations work consistently.

Ive already begun to order parts to build one of your V2 prototypes and build up a perf board, and can make changes to components as the code and tests progress. As a future plan, i would personally try and avoid using the MAF temp sensor, a pt1000 is magnitudes more accurate, and honestly ive found PT1000 sensors to be very cost effective for the performance they offer.

Thanks for the feedback and tips
